drivers/spi/spi_context.h

```diff
@@ -92,13 +92,14 @@ static inline void spi_context_lock(struct spi_context *ctx,
 				    void *callback_data,
 				    const struct spi_config *spi_cfg)
 {
-	if ((spi_cfg->operation & SPI_LOCK_ON) &&
-		(k_sem_count_get(&ctx->lock) == 0) &&
-		(ctx->owner == spi_cfg)) {
-			return;
+	bool already_locked = (spi_cfg->operation & SPI_LOCK_ON) &&
```

Comment:
Why not just adding a !() around existing tests?

Reply:
For the reason in the commit message:
`Extract the lock condition to a dedicated variable to make future
changes clearer.`
It's easier to invert the meaning of `already_locked` in your head than 3 checks ANDed together